src/include/uapi/ipc.h

```diff
@@ -495,20 +495,26 @@ struct sof_ipc_host_buffer {
 	uint32_t offset;
 } __attribute__((packed));
 
+struct sof_ipc_be_dma_params {
+	uint32_t be_count;
+	uint32_t be_comp_id[SOF_IPC_MAX_CHANNELS];
+	uint32_t be_dma_ch[SOF_IPC_MAX_CHANNELS];
+} __attribute__((packed));
+
 struct sof_ipc_stream_params {
 	struct sof_ipc_host_buffer buffer;
 	enum sof_ipc_stream_direction direction;
 	enum sof_ipc_frame frame_fmt;
 	enum sof_ipc_buffer_format buffer_fmt;
 	uint32_t host_dma_ch;
-	uint32_t be_dma_ch;
 	uint32_t rate;
 	uint32_t channels;
 	uint32_t sample_valid_bytes;
 	uint32_t sample_container_bytes;
 	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
 	enum sof_ipc_chmap chmap[SOF_IPC_MAX_CHANNELS];	/* channel map */
+	struct sof_ipc_be_dma_params be_dma_params;
```

Comment:
The design here is broken. If there has to be a relationship between host GW DMA channel and link DMA channel for HDA analog on Intel only platforms then this should be done by the CAVS platform code in the FW (and reuse the existing stream_id) and not via generic IPC from the driver.

Reply:
Hi Liam,  you mean these information is not generic,  can I add a private data array and convert it to internal data structure in intel platform code?

Thanks!

Reply:
I mean that choosing the link DMA channels is very platform specific and should be done in platform/cavs based on the stread_id (host DMA channel) being passed in.